jina/serve/gateway.py

```diff
@@ -0,0 +1,45 @@
+import abc
+import argparse
+from typing import TYPE_CHECKING, Optional
+
+from jina.jaml import JAMLCompatible
+from jina.logging.logger import JinaLogger
+from jina.serve.streamer import GatewayStreamer
+
+__all__ = ['BaseGateway']
+
+
+class BaseGateway(JAMLCompatible):
+    """
+    The base class of all custom Gateways, can be used to build a custom interface to a Jina Flow that supports
+    gateway logic
+
+    :class:`jina.Gateway` as an alias for this class.
+    """
+
+    def __init__(
+        self,
+        streamer: Optional[GatewayStreamer] = None,
+        args: 'argparse.Namespace' = None,
+        logger: 'JinaLogger' = None,
```

Comment:
same for logger if it JAML Compatible, it should only have args

Reply:
it should be jaml compatible now